### PR TITLE
Fix default behaviour for init function in `AppriseNotifier`

### PR DESCRIPTION
**Pull request by rrr2rrr**
---

Fixed default behaviour for init `send_apprise_notification` from `airflow.providers.apprise.notifications.apprise`

**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by eladkal** [comment]
Tests are failing

File: airflow/providers/apprise/hooks/apprise.py [hunk 1]
```diff
@@ -71,7 +72,7 @@ def notify(
         title: str | None = None,
         notify_type: NotifyType = NotifyType.INFO,
         body_format: NotifyFormat = NotifyFormat.TEXT,
-        tag: str | Iterable[str] | None = None,
+        tag: str | Iterable[str] = MATCH_ALL_TAG,
```
Comment (dirrao) [inline airflow/providers/apprise/hooks/apprise.py]: is this braking change?

Comment (rrr2rrr) [reply airflow/providers/apprise/hooks/apprise.py]: Yes! Passing None will not send the notification at all. Our team struggled for 2 hours to find this bug.

Comment (rrr2rrr) [reply airflow/providers/apprise/hooks/apprise.py]: How it should work according to the Apprise:

>         if the tag is specified (either a string or a set/list/tuple
>         of strings), then only the notifications flagged with that
>         tagged value are notified.  By default, all added services
>         are notified (tag=MATCH_ALL_TAG)

**Review by iakov-kaiumov — APPROVED** [review]

**Comment by eladkal** [comment]
@rrr2rrr can you please fix the test?
`FAILED tests/providers/apprise/notifications/test_apprise.py::TestAppriseNotifier::test_notifier - AssertionError: expected call not found.`

**Pull request closed without merging**